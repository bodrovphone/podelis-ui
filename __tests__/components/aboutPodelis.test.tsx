import React from 'react';
import '@testing-library/jest-dom/extend-expect';

import { render } from '../config/test-utils';
import AboutPodelis from '../../components/aboutPodelis';

describe('AboutPodelis', () => {
  it('should render logo', () => {
    const { getByText } = render(<AboutPodelis />);

    const logo = getByText('P', { exact: true });

    expect(logo).toBeInTheDocument();
  });

  it('should render slogan', () => {
    const { getByText } = render(<AboutPodelis />);

    const slogan = getByText(/Podelis - Простой и удобный шаринг-сервис\./i);

    expect(slogan).toMatchSnapshot();
  });

  it('should render 3 social media logos', () => {
    const { container } = render(<AboutPodelis />);

    // the reason of this weird selector is that next Image component duplicates them for optimization purpose.
    const logos = container.querySelectorAll('img:not([aria-hidden])');

    expect(logos.length).toBe(3);
  });
});
