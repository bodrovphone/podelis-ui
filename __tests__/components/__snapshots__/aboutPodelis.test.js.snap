// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AboutPodelis should render slogan 1`] = `
<div
  class="styles__Content-sc-6rzojn-1 bDqzaL"
>
  Podelis - Простой и удобный шаринг-сервис.
  <br />
  Арендуй поблизости.
  <br />
  Пользуйся тем что нужно, не покупая.
  <br />
  Сдавай в аренду вещи и зарабатывай.
  <br />
</div>
`;
